### PR TITLE
feat(smtpmock): refact server and add auth, create command line client

### PR DESCRIPTION
**Pull request by richardlt**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11925.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11931.0 ■   
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11931.1 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ■   
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11931.2 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11932.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.0 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ■   
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.1 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ■   
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.2 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.3 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.4 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ■   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ■   
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.5 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ✔ 
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.6 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ■   
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ✔ 
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.7 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ■   
  * Lint UI Translate ■   
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.8 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ■   
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ✔ 
  * Test SDK ■   
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.9 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ■   
  * Docs ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.10 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ■   
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.11 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests ✔ 
  * Standalone Tests ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.12 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ■   
  * Docs ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.13 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ■   
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ■   
  * Test CLI ■   
  * Test Engine ✔ 
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.14 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.15 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.16 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✘  
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.17 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests ✔ 
  * Standalone Tests ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11933.18 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests ✔ 
  * Standalone Tests ■

File: tools/smtpmock/server/api.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		AuthScheme: "Bearer",
+		Validator: func(key string, c echo.Context) (bool, error) {
+			if _, err := CheckSessionToken(key); err != nil {
+				return false, nil
+			}
+			return true, nil
+		},
+	}))
+
+	{ // sub routes for /messages
```
Comment (bnjjj) [inline tools/smtpmock/server/api.go]: don't understand what's the point of this block's creation

File: tools/smtpmock/server/store.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	defer allMessagesMutex.Unlock()
+
+	list := allMessages[m.To]
+	list = append([]smtpmock.Message{m}, list...)
+	allMessages[m.To] = list
+	messagesCounter++
+}
+
+func StoreGetMessages() []smtpmock.Message {
+	ms := []smtpmock.Message{}
```
Comment (bnjjj) [inline tools/smtpmock/server/store.go]:
```suggestion
	ms := make([]smtpmock.Message, 0, len(allMessages))
```

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11936.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11936.1 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ■   
  * Build OS-Ansible-Inventory ■   
  * Build SMTPMock ■   
  * Lint UI Translate ■   
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11936.2 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11938.0 ■   
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ■   
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11938.1 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11940.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Pull request merged**